src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -75,6 +75,53 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     apiOperations
   )
 
+  /** Schedules method for fetching baking rights */
+  system.scheduler.schedule(lorreConf.rights.initDelay, lorreConf.rights.interval)(writeFutureRights())
+
+  /** Fetches future baking and endorsing rights to insert it into the DB */
+  def writeFutureRights(): Unit = {
+    import cats.implicits._
+
+    implicit val mat = ActorMaterializer()
+    logger.info("BER Fetching future baking and endorsing rights")
+    val blockHead = tezosNodeOperator.getBlockHead()
+    val brLevelFut = apiOperations.fetchMaxBakingRightsLevel()
+    val erLevelFut = apiOperations.fetchMaxEndorsingRightsLevel()
+
+    (blockHead, brLevelFut, erLevelFut).mapN { (head, brLevel, erLevel) =>
+      val headLevel = head.data.header.level
+      val rightsStartLevel = math.max(brLevel, erLevel) + 1
+      logger.info(s"BER Current levels block: $headLevel baking rights: $brLevel endorsing rights: $erLevel")
+
+      val length = DatabaseConversions
```

Comment:
Maybe some comments might help to follow what's been done?

Reply:
added a comment

Reply:
To be honest I still have yet not clear why the parameters are computed like that and why those are the levels to fetch.

Beware, I'm not saying that the computations are wrong. It's simply that I'm not able to understand them, so I trust you did reason correctly on this and will take time later if needed to properly understand the logic.

Reply:
it was described in the ticket